Seneca-CDOT/telescope
# Update telescope_profiles and feeds tables

## Issue This PR Addresses
Part of #2860

This updates `feeds` table schema in preparation for migrating [Planet feed list](https://wiki.cdot.senecacollege.ca/wiki/Planet_CDOT_Feed_List) to Supabase. 

Schema
|  Field | Description  | 
|---|---|
|  url |   text -  primary key - unique feed url|
|   user_id |  text - optional foreign key - A register can claim an existing feed when they register for Telescope|
|  author_name |  text - author name get from Planet, will be unused when the feed is linked with a user |
|  html_url |  text optional - Actually blog URL of the RSS feed |
|  type |  'blog' \| 'youtube' \| 'twitch' |
|  flagged |  boolean |
|  Invalid |  boolean |

## Steps to test the PR

1. Pick up the new schema script
   ```
    docker-compose stop db && docker-compose rm db
    docker-compose up db -d
   ```
2. Rebuild the SSO service
   ```
    docker-compose build sso
   ```
3. Start all services
   ``` 
    pnpm  dev
    pnpm services:start
   ```
4. Register Telescope user at http://localhost:8000/signup
5. If a feed entered is already associated with a user, registration should fail
